runtime/lua/nvim/health.lua

```diff
@@ -54,9 +54,16 @@ local function check_config()
   health.start('Configuration')
   local ok = true
 
-  local vimrc = (
-    empty(vim.env.MYVIMRC) and vim.fn.stdpath('config') .. '/init.vim' or vim.env.MYVIMRC
-  )
+  local init_lua_path = vim.fn.stdpath('config') .. '/init.lua'
+  local init_vim_path = vim.fn.stdpath('config') .. '/init.vim'
+  if filereadable(init_lua_path) and filereadable(init_vim_path) then
+    health.warn('You can have either Vimscript ("init.vim") or Lua ("init.lua"), but not both.')
+  end
+
+  local init_path = filereadable(init_lua_path) and init_lua_path or init_vim_path
+
+  local vimrc = (empty(vim.env.MYVIMRC) and init_path or vim.env.MYVIMRC)
```

Comment:
should probably default to init.lua rather than init.vim, for messaging.
```suggestion
  local vimrc = (empty(vim.env.MYVIMRC) and init_lua_path or vim.env.MYVIMRC)
```